cloudinit/util.py

```diff
@@ -2738,20 +2740,35 @@ def parse_mount(path, get_mnt_opts=False):
         # continue finding the real device like '/dev/da0'.
         # this is only valid for non zfs file systems as a zpool
         # can have gpt labels as disk.
+        # It also doesn't really make sense for NFS.
         devm = re.search("^(/dev/.+)[sp]([0-9])$", devpth)
         if not devm and is_FreeBSD() and fs_type not in ["zfs", "nfs"]:
-            devpth = get_freebsd_devpth(path)
+            # don't duplicate the effort of finding the mountpoint in
+            # ``get_freebsd_devpth()`` by passing it the ``path``
+            # instead only resolve the ``devpth``
+            devpth = get_freebsd_devpth(devpth)
+        match_devpth = devpth
 
         if match_mount_point == path:
             break
 
-    if mount_point and path in mount_point:
+    if match_mount_point and match_mount_point in path:
         if get_mnt_opts:
-            if devpth and fs_type and match_mount_point and mount_options:
-                return (devpth, fs_type, match_mount_point, mount_options)
+            if (
+                match_devpth
+                and fs_type
+                and match_mount_point
+                and mount_options
+            ):
+                return (
+                    match_devpth,
+                    fs_type,
+                    match_mount_point,
+                    mount_options,
+                )
         else:
-            if devpth and fs_type and match_mount_point:
-                return (devpth, fs_type, match_mount_point)
+            if match_devpth and fs_type and match_mount_point:
+                return (match_devpth, fs_type, match_mount_point)
     return None
```

Comment:
need to check the logic tables to see if after returning early we'd still need this `return None`

Reply:
A `return None` is implicit in Python functions, so we really don't even need this one.

Reply:
aye, i thought it was *our* preferred style